gossip/gossip.go

```diff
@@ -554,15 +564,16 @@ func (g *Gossip) bootstrap(stopper *stop.Stopper) {
 // is notified via the stalled conditional variable.
 func (g *Gossip) manage(stopper *stop.Stopper) {
 	stopper.RunWorker(func() {
-		// Loop until closed and there are no remaining outgoing connections.
 		for {
 			select {
 			case <-stopper.ShouldStop():
 				return
 			case c := <-g.disconnected:
 				g.doDisconnected(stopper, c)
-			case <-time.After(g.jitteredInterval(checkInterval)):
-				g.doCheckNetwork(stopper)
+			case nodeID := <-g.tighten:
+				g.tightenNetwork(stopper, nodeID)
+			case <-time.After(g.jitteredInterval(cullInterval)):
+				g.cullNetwork()
 			case <-time.After(g.jitteredInterval(stallInterval)):
```

Comment:
I must be missing something, but for the life of me I can't figure out how these two `time.After` channels ever allowed the larger delay (`checkInterval` before, now `cullInterval`) to trigger. Based on my understanding, each iteration in the for loop would create new timers in the select block, and the timer with the shorter duration would always trigger before the other one. The duration jittering isn't strong enough to allow the larger delay to ever be smaller than the jittered shorter delay, so I can't see it ever triggering `doCheckNetwork` or `cullNetwork`. Here's a [Go playground](https://play.golang.org/p/_nPsNNSEAm) to demonstrate what I think would result, along with an [example](https://play.golang.org/p/jS-1Q0EDD-) of using a `time.Ticker` to achieve what I believe to be the desired interaction.